core/arch/x86/decode.c

```diff
@@ -708,6 +708,135 @@ read_vex(byte *pc, decode_info_t *di, byte instr_byte,
     return pc;
 }
 
+/* Given the potential first evex byte at pc, reads any subsequent evex
+* bytes (and any prefix bytes) and sets the appropriate prefix flags in di.
+* Sets info to the entry for the first opcode byte, and pc to point past
+* the first opcode byte.
+*/
+static byte *
+read_evex(byte *pc, decode_info_t *di, byte instr_byte,
+          const instr_info_t **ret_info INOUT)
+{
+
+    const instr_info_t *info;
+    byte prefix_byte = 0, evex_pp = 0;
+    ASSERT(ret_info != NULL && *ret_info != NULL);
+    info = *ret_info;
+
+    CLIENT_ASSERT(info->type == EVEX_PREFIX_EXT, "internal evex decoding error");
+
+    /* If 64-bit mode, P[3:2] are 0 and P[10] is 1 then this is evex */
+    if (X64_MODE(di) && !TEST(0xC, *pc) && TEST(0x04, *(pc + 1))) {
```

Comment:
Should be similar to VEX. Either di mode (0x62 is always EVEX) or (||) test modrm MOD for 0b11, which is invalid for bound and therefore EVEX.

The bits P[3:2] and P[10] should be 0b00 and 0b1 respectively, otherwise return #UD similar as read_vex does it.

Reply:
Please resolve conversations if my changes meet requirements.